test/unit/org/apache/cassandra/metrics/DecayingEstimatedHistogramReservoirTest.java

```diff
@@ -604,6 +614,115 @@ public void testSize()
+         * The load has 95% of 42, and 5% of the time it's 1109. Despite that the histogram may be convinced for a long time
+         * that p99 is 42 or that p50 is 1109.
+         * The reason may be seen in the snapshot dump, where after rescale the bucket values may get
+         * very big due to the race condition and too big weight of the inserted samples.
+         * The values were picked to match bucket boundaries, but that's only for aesthetics.
+         * <p>
+         * In production the rescale happens every 30 minutes. In this test time flies roughly 1000 times faster to
+         * hit the condition in a reasonable time.
+         */
+        @Ignore("This test exposes a specific CASSANDRA-19365 race condition; it doesn't make sense to run it in CI")
+        @Test
+        public void testConcurrentUpdateAndRescale() throws InterruptedException
+        {
+            int maxTestDurationMillis = 60_000;
+            AtomicBoolean stop = new AtomicBoolean(false);
+            AtomicBoolean failed = new AtomicBoolean(false);
+            int NUM_UPDATE_THREADS = 60;
+            TestClock clock = new TestClock();
+
+            DecayingEstimatedHistogramReservoir histogram = new DecayingEstimatedHistogramReservoir(clock);
+
+            ExecutorService executors = Executors.newFixedThreadPool(2 + NUM_UPDATE_THREADS);
+            // NUM_UPDATE_THREADS threads updating the histogram
+            for (int i = 0; i < NUM_UPDATE_THREADS; i++)
+            {
+                executors.submit(() -> {
+                    while (!stop.get())
+                    {
+                        // a mischievous usage pattern to quickly trigger the
+                        // CASSANDRA-19365 race condition;
+                        // the load has 95% of 42, and only 5% of the time it's 1109
+                        // and yet, the histogram may be convinced for a long time that
+                        // the p99 is 42 or that the p50 is 1109
+                        for (int sampleIdx = 0; sampleIdx < 900; sampleIdx++)
+                        {
+                            histogram.update(42);
+                        }
+                        for (int sampleIdx = 0; sampleIdx < 50; sampleIdx++)
+                        {
+                            // add some noise so that low value samples do not race with the same likelyhood as the high value samples
+                            Uninterruptibles.sleepUninterruptibly(1, MILLISECONDS);
+                            histogram.update(1109);
+                        }
+                    }
+                });
+            }
+            // clock update thread
+            executors.submit(() -> {
+                while (!stop.get())
+                {
+                    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.MILLISECONDS);
+                    // x1000 speedup so that we hit rescale interval every 30 minutes / 1000 = 1.8s
+                    clock.addMillis(1000);
```

Comment:
It's better to use this value in relation to the `LANDMARK_RESET_INTERVAL_IN_NS` itself.